core/controllers/suggestion.py

```diff
@@ -40,21 +40,107 @@
 class SuggestionHandler(base.BaseHandler):
     """"Handles operations relating to suggestions."""
 
+    POST_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+    URL_PATH_ARGS_SCHEMAS = {}
+    HANDLER_ARGS_SCHEMAS = {
+        'POST': {
+            'suggestion_type': {
+                'schema': {
+                    'type': 'basestring',
+                    'choices': feconf.SUGGESTION_TYPE_CHOICES
+                }
+            },
+            'target_type': {
+                'schema': {
+                    'type': 'basestring',
+                    'choices': feconf.SUGGESTION_TARGET_TYPE_CHOICES
+                }
+            },
+            'target_id': {
+                'schema': {
+                    'type': 'basestring'
+                }
+            },
+            'target_version_at_submission': {
+                'schema': {
+                    'type': 'int',
+                    'validators': [{
+                        'id': 'is_at_least',
+                        # Target version must be greater than 1.
+                        'min_value': 1
+                    }]
+                }
+            },
+            'change': {
+               'schema': {
+                   'type': 'dict',
+                   'properties': [
+                       {
+                            'name': 'cmd',
+                            'schema': {
+                                'type': 'basestring'
+                            }
+                       },
+                       {
+                            'name': 'state_name',
+                            'schema': {
+                                'type': 'basestring'
+                            }
+                       },
+                       {
+                            'name': 'content_id',
+                            'schema': {
+                                'type': 'basestring'
+                            }
+                       },
+                       {
+                            'name': 'language_code',
+                            'schema': {
+                                'type': 'basestring'
+                            },
+                            'default': None
```

Comment:
You can probably use similar code as in https://github.com/oppia/oppia/pull/13968#discussion_r721748102